script.sh

```diff
@@ -42,54 +42,41 @@ case "${INPUT_SETUP_METHOD}" in
     ;;
 esac
 
-# Install mypy if needed.
-if [[ "${SETUP}" == "true" ]] ; then
-  echo '::group:: Installing mypy ...  https://github.com/python/mypy'
+echo '::group:: Install mypy (if needed)'
+
+if [[ "${SETUP}" == "true" ]] || ! mymy --version ; then
   echo "Execute setup: ${INPUT_SETUP_COMMAND}"
   ${INPUT_SETUP_COMMAND}
-  echo '::endgroup::'
 fi
 
+echo '::endgroup::'
 echo '::group:: Prepare reviewdog/mypy'
-# Version output.
+
 echo "Execute command and version: ${INPUT_EXECUTE_COMMAND}"
 ${INPUT_EXECUTE_COMMAND} --version
 
 export REVIEWDOG_GITHUB_API_TOKEN="${INPUT_GITHUB_TOKEN}"
 
 # safe extract files/dirs
 TARGETS_LIST="${INPUT_TARGET:-.}"
+
 echo '::endgroup::'
+echo '::group:: Running mypy with reviewdog 🐶 ...'
 
-# pre-run(missing stub detect)
 if [[ "${INPUT_INSTALL_TYPES}" == "true" ]] ; then
-  echo '::group:: Installing types'
-  echo 'Pre-run and detect missing stubs'
-  # shellcheck disable=SC2086
-  mypy_check_output="$(${INPUT_EXECUTE_COMMAND}   \
-                            ${TARGETS_LIST} 2>&1  \
-                            )" || mypy_exit_val="$?"
-  # discard result
-  echo 'Install types'
-  ${INPUT_EXECUTE_COMMAND} --install-types --non-interactive
-  echo '::endgroup::'
+  INPUT_MYPY_FLAGS="${INPUT_MYPY_FLAGS} --install-types --non-interactive"
```

Comment:
This part is bad because mypy (certainly at least in older versions) needs to cache the results of one run to run install-types.

https://mypy.readthedocs.io/en/stable/command_line.html